### PR TITLE
[CLI] Support shorthand for loggers

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/10574

### Does your PR introduce any breaking changes? If yes, please list them.

Previous calls with `--trainer.logger=pytorch_lightning.loggers.WandbLogger` will need to update to `--trainer.logger=WandbLogger`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @awaelchli @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy @carmocca @mauvilsa

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/utilities/cli.py [hunk 1]
```diff
@@ -64,9 +64,16 @@ def __call__(self, cls: Type, key: Optional[str] = None, override: bool = False)
...
     def register_classes(self, module: ModuleType, base_cls: Type, override: bool = False) -> None:
         """This function is an utility to register all classes from a module."""
-        for _, cls in inspect.getmembers(module, predicate=inspect.isclass):
-            if issubclass(cls, base_cls) and cls != base_cls:
-                self(cls=cls, override=override)
+        for cls in self.get_members(module, base_cls):
+            self(cls=cls, override=override)
+
+    @staticmethod
+    def get_members(module: ModuleType, base_cls: Type) -> Generator[Type, None, None]:
```
Comment (tchaton) [inline pytorch_lightning/utilities/cli.py:71]: Neat !

File: pytorch_lightning/utilities/cli.py [hunk 2]
```diff
@@ -579,6 +586,7 @@ def add_core_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
         """Adds arguments from the core classes to the parser."""
         parser.add_lightning_class_args(self.trainer_class, "trainer")
         parser.set_choices("trainer.callbacks", CALLBACK_REGISTRY.classes, is_list=True)
+        parser.set_choices("trainer.logger", tuple(_Registry.get_members(pl.loggers, pl.loggers.LightningLoggerBase)))
```
Comment (tchaton) [inline pytorch_lightning/utilities/cli.py:589]: Can we create a proper registry for logger ?

**Review by tchaton — CHANGES_REQUESTED** [review]
Mind adding a LOGGER_REGISTRY and then all good :)

File: pytorch_lightning/utilities/cli.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/utilities/cli.py:589]: I don't agree with that. I'ts adding bloat considering there are no logger implementations outside of the project.

I'm totally open to adding it when the community requests it, but adding it is unrelated to supporting shorthand.

Comment (tchaton) [reply pytorch_lightning/utilities/cli.py:589]: The point of adding a registry is for users to easily add any external library such as the aim logger or any other and benefit from shorthand without adding them within Lightning. IMO, this needs to be added :)

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]

**Comment by ia-davidpichler** [comment]
Is there a way to set the defaults for a custom logger's arguments in code? e.g.
```
class MyCLI(LightningCLI):
    def add_arguments_to_parser(self, parser) -> None:
        parser.set_defaults({
            "trainer.logger": "TensorBoardLogger",
            "trainer.logger.save_dir": "./logdir/"
        })
```

**Comment by mauvilsa** [comment]
> Is there a way to set the defaults for a custom logger's arguments in code?

You can do it like:

```python
        parser.set_defaults({
            "trainer.logger": {
                "class_path": "pytorch_lightning.loggers.TensorBoardLogger",
                "init_args": {
                    "save_dir": "./logdir/"
                }
            }
        })
```

**Pull request merged**